### ISSUE TITLE
TimeAmount constructors generate suboptimal code.

### ISSUE THREAD

**Issue by Lukasa** [apple/swift-nio#1249]
The `TimeAmount` struct has several static constructors to generate a `TimeAmount` in many units. For example:

```swift
    public static func seconds(_ amount: Int64) -> TimeAmount {
        return TimeAmount(amount * 1000 * 1000 * 1000)
    }
```

Unfortunately this leads to suboptimal codegen because Swift needs an overflow check at each multiplication stage.

We can improve this by adding some parentheses:

```swift
    public static func seconds(_ amount: Int64) -> TimeAmount {
        return TimeAmount(amount * (1000 * 1000 * 1000))
    }
```

This will lead to much better codegen.

**Comment by Lukasa** [comment]
See: https://swift.godbolt.org/z/QP7S6N

**Comment by weissi** [comment]
fixed by #1250
